src/cli/cache.rs

```diff
@@ -25,28 +24,18 @@ impl<K: Hash + Eq, V: Clone> NativeChronoCache<K, V> {
     NativeChronoCache { data: Arc::new(RwLock::new(TtlCache::new(CACHE_CAPACITY))) }
   }
 }
+
 #[async_trait::async_trait]
 impl<K: Hash + Eq + Send + Sync, V: Clone + Send + Sync> Cache for NativeChronoCache<K, V> {
   type Key = K;
   type Value = V;
   #[allow(clippy::too_many_arguments)]
-  async fn set<'a>(&'a self, key: K, value: V, ttl: NonZeroU64) -> Result<V> {
+  async fn set<'a>(&'a self, key: K, value: V, ttl: NonZeroU64) -> Option<V> {
     let ttl = Duration::from_millis(ttl.get());
-    self
-      .data
-      .write()
-      .unwrap()
-      .insert(key, value, ttl)
-      .ok_or(anyhow!("unable to insert value"))
+    self.data.write().unwrap().insert(key, value, ttl)
   }
 
-  async fn get<'a>(&'a self, key: &'a K) -> Result<V> {
-    self
-      .data
-      .read()
-      .unwrap()
-      .get(key)
-      .cloned()
-      .ok_or(anyhow!("key not found"))
```

Comment:
There are many changes being done here, can we add a test case.

Reply:
added unit test for `NativeChronoCache`